interop_task/interop_task.md

```diff
@@ -0,0 +1,123 @@
+executed in-order w.r.t the other command groups on the same queue.
+Simultaneously, we guarantee that this operation is performed
+asynchronously w.r.t to the user-thread (therefore, enabling the user
+thread to continue submitting command groups).
+Other command groups enqueued in the same or different queues
+can be executed following the sequential consistency by guaranteeing the
+satisfaction of the requisites of this command group.
+
+The possibility of enqueuing host tasks on SYCL queues also enables the
+runtime to perform further optimizations when available.
+For example, a SYCL runtime may decide to map / unmap instead of copy operations,
+or performing asynchronous transfers while data is being computed.
+
+### cl::sycl::codeplay::handler
+
+```cpp
+namespace cl {
+namespace sycl {
+namespace codeplay {
+
+class handler : public cl::sycl::handler {
+ private:
+  // implementation defined constructor
+  handler(__unspecified__);
+
+ public:
+  /* "Manually" enqueue a kernel */
+  template <typename FunctorT>
+  void interop_task(FunctorT hostFunction);
+};
+}  // namespace codeplay
+}  // namespace sycl
+}  // namespace cl
+```
+
+### codeplay::handler::interop_task
+
+The `interop_task` allow the user to execute a task of the native host.
+Unlike `single_task`, `parallel_for` and `parallel_for_work_group`, the `interop_task` do not enqueue a kernel on the device but allow the user to execute a custom action when the prerequisites are satisfied on the device associate with the queue.
+The functor passed to the `interop_task` takes as input a const reference to a `cl::sycl::codeplay::interop_handle` which can be used to retrieve underlying OpenCL objects relative to the execution of the task.
+
+It is not allowed to allocate new SYCL object inside an `interop_task`.
+It is the user responsibilities to ensure all asynchronous executions using SYCL provided resources finished before returning from the `interop_task`.
+
+## Accessing Underlying OpenCL Object
+
+We introduce the `interop_handle` class which provide access to underlying OpenCL objects during the execution of the `interop_task`.
+
+The interface of the `interop_handle` is defined as follow:
+```cpp
+namespace cl {
+namespace sycl {
+namespace codeplay {
+
+class interop_handle {
+ private:
+  // implementation defined constructor
+  interop_handle(__unspecified__);
+
+ public:
+  /* Return the context */
+  cl_context get_context() const;
+
+  /* Return the device id */
+  cl_device_id get_device() const;
+
+  /* Return the command queue associated with this task */
+  cl_command_queue get_queue() const;
+
+  /*
+    Returns the underlying cl_mem object associated with the accessor
+  */
+  template <typename dataT, int dimensions, access::mode accessmode,
+            access::target accessTarget,
+            access::placeholder isPlaceholder>
+  cl_mem get(const accessor<dataT, dimensions, accessmode, access::target accessTarget, access::placeholder isPlaceholder>&) const;
```

Comment:
In SYCL there is a convention to use the name `get` as a way to return the interop object associated with the object calling `get`, so in this case, I think we should use a different name, perhaps `get_buffer`?